packages/server-wallet/src/protocols/__test__/defunder.test.ts

```diff
@@ -0,0 +1,119 @@
+import {defaultTestConfig} from '../..';
+import {testKnex} from '../../../jest/knex-setup-teardown';
+import {MockChainService} from '../../chain-service';
+import {DBAdmin} from '../../db-admin/db-admin';
+import {createLogger} from '../../logger';
+import {AdjudicatorStatusModel} from '../../models/adjudicator-status';
+import {Channel} from '../../models/channel';
+import {Funding} from '../../models/funding';
+import {Store} from '../../wallet/store';
+import {TestChannel} from '../../wallet/__test__/fixtures/test-channel';
+import {Defunder} from '../defunder';
+
+const testChannelObj = TestChannel.create({
+  aBal: 5,
+  bBal: 3,
+  finalFrom: 4,
+});
+let store: Store;
+
+beforeEach(async () => {
+  await DBAdmin.truncateDataBaseFromKnex(testKnex);
+
+  store = new Store(
+    testKnex,
+    defaultTestConfig().metricsConfiguration.timingMetrics,
+    defaultTestConfig().skipEvmValidation,
+    '0'
+  );
+});
+
+describe('Direct channel defunding', () => {
+  it('Channel is defunded via submission of withdrawAndConlcude transaction', async () => {
+    // Channel has yet to be concluded
+    await testChannelObj.insertInto(store, {
+      participant: 0,
+      states: [3, 4],
+      funds: 8,
+    });
+    const chainService = new MockChainService();
+    const spy = jest.spyOn(chainService, 'concludeAndWithdraw');
+    const logger = createLogger(defaultTestConfig());
+
+    let channel = await Channel.forId(testChannelObj.channelId, testKnex);
+    const defunder = new Defunder(store, chainService, logger);
+
+    testKnex.transaction(async tx => {
+      // Defunder should take no actions as there is no conclusion proof.
+      expect(await defunder.crank(channel, tx)).toEqual({
+        didSubmitTransaction: false,
+        isChannelDefunded: false,
+      });
+      expect(spy).not.toHaveBeenCalled();
+
+      // Defunder should submit a transaction since there is a conclusion proof.
+      const state = testChannelObj.wireState(5);
+      channel = await store.addSignedState(channel.channelId, state, tx);
```

Comment:
(out-of-PR-scope?): This smells a bit off to me, as we're using some store feature to set up the DB state for this test. I can't exactly say what's wrong with this.

`TestChannel` instances have a `insertInto` method, which seems useful. Should there be a `testChannelObj.setVars` method? I don't know if that's more maintainable or less maintainable.

Reply:
`await testChannel.insertInto(store, {states: [5]})` would actually work in this case except that the transaction object must be used for all database operations that `insertInto` invokes.